lib/services/role.go

```diff
@@ -1525,6 +1540,26 @@ func (l *windowsLoginMatcher) Match(role types.Role, typ types.RoleConditionType
 	return false, nil
 }
 
+type kubernetesClusterLabelMatcher struct {
+	clusterLabels map[string]string
+}
+
+// NewKubernetesClusterLabelMatcher creates a RoleMatcher that checks whether the role's
+// matches Kubernetes service labels.
+func NewKubernetesClusterLabelMatcher(clustersLabels map[string]string) RoleMatcher {
+	return &kubernetesClusterLabelMatcher{clusterLabels: clustersLabels}
+}
+
+// Match matches a Kubernetes cluster labels against a role.
+func (l *kubernetesClusterLabelMatcher) Match(role types.Role, typ types.RoleConditionType) (bool, error) {
+	if len(l.clusterLabels) == 0 {
+		// If a kubernetes cluster doesn't have any labels match the role.
+		return true, nil
+	}
```

Comment:
I am not sure this is a safe default. Say, Kubernetes cluster does not have any labels but the role allows `env: prod`. Does the cluster match the labels in this case or not? I'd err on the side of caution and not match in this case. I think the current behavior is that only wildcard selector matches empty labels.

Reply:
Yes That how CheckAccess to kube resource works. But this is a `CheckKubeGroupsAndUsers` logic where the impersonate Kube groups and users are returned. 

Before this fix if a kube service don't contain any label the union of KubeUsers and KubeGrups for user's roles are returned. I don't want to change this behavior because it will require to our customer to align current kube config.

Reply:
@smallinsky So if that's how CheckAccess works, then supposedly if the cluster does not have labels then we won't allow access to it anyway. So why is this logic here needed then if the access will be denied anyway?

Reply:
Like you mention the access will be denied by CheckAccess. I wanted to keep old behaviour but as you pointed out it doesn't make sens at this point because it will never happen. Removed Thanks.